sys/net/gnrc/doc.txt

```diff
@@ -0,0 +1,169 @@
+/**
+ * @defgroup    net_gnrc Generic (gnrc) network stack.
+ * @ingroup     net
+ * @brief       RIOT's modular default IP network stack.
+ *
+ * # About
+ *
+ * This module is currently the default network stack of RIOT and includes many components
+ * ranging from a simple mac protocol (@ref net_gnrc_nomac) through a fully-featured IPv6
+ * implementation (@ref net_gnrc_ipv6) with 6LowPAN extensions (@ref net_gnrc_sixlowpan) to
+ * an udp implementation (@ref net_gnrc_udp) and the Routing Protocol for Low Power and Lossy
+ * Networks (@ref net_gnrc_rpl).
+ *
+ * A list of all features contained in the @ref net_gnrc is available in the `Modules` section above.
+ *
+ * # Integration into RIOT-OS
+ *
+ * The integration of the @ref net_gnrc into RIOT-OS is handled by two modules:
+ * - @ref net_gnrc_conn
+ * - @ref net_gnrc_netdev2
+ *
+ * The @ref net_gnrc_conn implements the API to the application layer (@ref net_conn),
+ * while the @ref net_gnrc_netdev2 implements the API to communicate with low-level
+ * network device drivers (@ref drivers_netdev_netdev2).
+ *
+ * # Architecture
+ *
+ * ![GNRC Network Stack](riot-gnrc.svg)
+ * Each layer of the network stack resembles a stand-alone thread that runs with a thread priority
```

Comment:
Proposal:
"Each layer of the network stack runs in its own thread. Each lower layer thread runs with a higher priority than any upper layer thread.`